azurerm/data_source_netapp_pool.go

```diff
@@ -0,0 +1,76 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	aznetapp "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/netapp"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmNetAppPool() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmNetAppPoolRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppPoolName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppAccountName,
+			},
+
+			"service_level": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"size_in_4_tb": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmNetAppPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.PoolClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	accountName := d.Get("account_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: NetApp Pool %q (Resource Group %q) was not found", name, resourceGroup)
+		}
+		return fmt.Errorf("Error reading NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", name)
+	d.Set("account_name", accountName)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if poolProperties := resp.PoolProperties; poolProperties != nil {
+		d.Set("service_level", poolProperties.ServiceLevel)
+		d.Set("size_in_4_tb", *poolProperties.Size/4398046511104)
```

Comment:
This could be nil so we should check that it's not before addressing it
```suggestion
if poolProperties.Size != nil {
		d.Set("size_in_4_tb", *poolProperties.Size/4398046511104)
}
```

Reply:
Fixed